probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -1198,6 +1277,38 @@ impl MemoryInterface for Armv7a<'_> {
         })
     }
 
+    fn write(&mut self, address: u64, data: &[u8]) -> Result<(), Error> {
+        self.halted_access(|core| {
+            let len = data.len();
+            let start_extra_count = ((4 - (address % 4) as usize) % 4).min(len);
+            let end_extra_count = (len - start_extra_count) % 4;
+            assert!(start_extra_count < 4);
+            assert!(end_extra_count < 4);
+
+            if start_extra_count != 0 || end_extra_count != 0 {
```

Comment:
I don't understand this. Why does a "write a list of single bytes" function have to care about alignment? This function shouldn't return a MemoryNotAlignedError

Reply:
This was copied almost directly from https://github.com/probe-rs/probe-rs/blob/5d41d234ff64656ff93d1fc33e5f811f438f5ff0/probe-rs/src/memory/mod.rs#L280-L291, except I removed the check for `self.supports_8bit_transfers()` since it unconditionally returns `false`.

Why does the original behave like that, and why isn't it an issue?

Reply:
The documentation for the `write()` trait does say:

```rust
    /// If the target does not support 8-bit aligned access, and `address` is not
    /// aligned on a 32-bit boundary, this function will return a [`Error::MemoryNotAligned`] error.
```

Reply:
Eww okay let's stick with it then, but I wonder why that was done so.